config/v1/types_oauth.go

```diff
@@ -58,43 +62,52 @@ type TokenConfig struct {
 }
 
 const (
-	// LoginTemplateKey is the default key of the login template
+	// LoginTemplateKey is the key of the login template in a secret
 	LoginTemplateKey = "login.html"
-	// ProviderSelectionTemplateKey is the default key for the provider selection template
+	// ProviderSelectionTemplateKey is the key for the provider selection template in a secret
 	ProviderSelectionTemplateKey = "providers.html"
-	// ErrorsTemplateKey is the default key for the errors template
+	// ErrorsTemplateKey is the key for the errors template in a secret
 	ErrorsTemplateKey = "errors.html"
+
+	// BindPasswordKey is the key for the LDAP bind password in a secret
+	BindPasswordKey = "bindPassword"
+
+	// ClientSecretKey is the key for the oauth client secret data in a secret
+	ClientSecretKey = "clientSecret"
+
+	// HTPasswdDataKey is the key for the htpasswd file data in a secret
+	HTPasswdDataKey = "htpasswd"
 )
 
 // OAuthTemplates allow for customization of pages like the login page
 type OAuthTemplates struct {
-	// login is a reference to a secret that specifies a go template to use to render the login page.
-	// If a key is not specified, the key `login.html` is used to locate the template data.
+	// login is the name of a secret that specifies a go template to use to render the login page.
+	// The key "login.html" is used to locate the template data.
 	// If unspecified, the default login page is used.
 	// +optional
-	Login LocalSecretReference `json:"login,omitemtpy"`
+	Login string `json:"login,omitemtpy"`
```

Comment:
We want a specific type to reference a LocalSecret.  I can be flexible about creating a new type that has no `key` (though I find this particular change of questionable value), but an untyped string is not a suitable choice.